src/common.speech/SynthesisAdapterBase.ts

```diff
@@ -0,0 +1,630 @@
+                    connectionClosedEvent.statusCode === 1007 ? CancellationErrorCode.BadRequestParameters : CancellationErrorCode.ConnectionFailure,
+                    connectionClosedEvent.reason + " websocket error code: " + connectionClosedEvent.statusCode);
+            }
+        });
+    }
+
+    public static addHeader(audio: ArrayBuffer, format: AudioOutputFormatImpl): ArrayBuffer {
+        if (!format.hasHeader) {
+            return audio;
+        }
+        format.updateHeader(audio.byteLength);
+        const tmp = new Uint8Array(audio.byteLength + format.header.byteLength);
+        tmp.set(new Uint8Array(format.header), 0);
+        tmp.set(new Uint8Array(audio), format.header.byteLength);
+        return tmp.buffer;
+    }
+
+    public isDisposed(): boolean {
+        return this.privIsDisposed;
+    }
+
+    public dispose(reason?: string): void {
+        this.privIsDisposed = true;
+        if (this.privSessionAudioDestination !== undefined) {
+            this.privSessionAudioDestination.close();
+        }
+        if (this.privConnectionConfigurationPromise) {
+            this.privConnectionConfigurationPromise.onSuccessContinueWith((connection: IConnection) => {
+                connection.dispose(reason);
+            });
+        }
+    }
+
+    public connect(): void {
+        this.connectImpl().result();
+    }
+
+    public connectAsync(cb?: Callback, err?: Callback): void {
+        this.connectImpl().continueWith((promiseResult: PromiseResult<IConnection>) => {
+            try {
+                if (promiseResult.isError) {
+                    if (!!err) {
+                        err(promiseResult.error);
+                    }
+                } else if (promiseResult.isCompleted) {
+                    if (!!cb) {
+                        cb();
+                    }
+                }
+            } catch (e) {
+                if (!!err) {
+                    err(e);
+                }
+            }
+        });
+    }
+
+    public Speak(
+        text: string,
+        isSSML: boolean,
+        requestId: string,
+        successCallback: (e: SpeechSynthesisResult) => void,
+        errorCallBack: (e: string) => void,
+        audioDestination: IAudioDestination,
+    ): Promise<boolean> {
+
+        let ssml: string;
+
+        if (isSSML) {
+            ssml = text;
+        } else {
+            ssml = SpeechSynthesizer.buildSsml(text, this.privSynthesizerConfig.parameters);
+        }
+
+        if (this.speakOverride !== undefined) {
+            return this.speakOverride(ssml, requestId, successCallback, errorCallBack);
+        }
+
+        this.privSuccessCallback = successCallback;
+        this.privErrorCallback = errorCallBack;
+
+        this.privSynthesisTurn.startNewSynthesis(requestId, text, isSSML, audioDestination);
+
+        // Start the connection to the service. The promise this will create is stored and will be used by configureConnection().
+        this.connectImpl();
```

Comment:
You can just call fetchConnection since you don't need to do something expensive like getting a microphone in parallel.

Reply:
thx, I see. Will update.